static/js/settings_users.js

```diff
@@ -359,6 +365,28 @@ exports.on_load_success = function (realm_people_data) {
                     data.bot_owner_id = people.get_by_email(owner_select_value).user_id;
                 }
             } else {
+                var new_profile_data = [];
+                $("#user-info-form-modal .custom_user_field_value").each(function () {
+                    // Remove duplicate datepicker input element genearted flatpicker library
+                    if (!$(this).hasClass("form-control")) {
+                        new_profile_data.push({
+                            id: parseInt($(this).closest(".custom_user_field").attr("data-field-id"), 10),
+                            value: $(this).val(),
+                        });
+                    }
+                });
+                // Append user type field values also
+                for (var field_id in fields_user_pills) {
```

Comment:
We should use `forEach` or `_.each` as `for...in` has so many drawbacks (though I'm not sure whether they are applicable here too or not... but still we are more consistent in using `forEach` or `_.each`)